packages/scene-system/sdk/SceneRuntime.ts

```diff
@@ -533,7 +535,11 @@ export abstract class SceneRuntime extends Script {
       }
 
       try {
-        await this.runCode(source as any as string, { dcl })
+        const { Permissions } = await this.loadAPIs(['Permissions'])
+        const restrictedWebSocket = createWebSocket(Permissions)
+        const restrictedFetch = createFetch(Permissions)
+
+        await this.runCode(source as any as string, { dcl, WebSocket: restrictedWebSocket, fetch: restrictedFetch })
```

Comment:
I'd also patch the globalThis.fetch, keeping the original fetch in a variable for internal usage inside the SceneRuntime